### PR TITLE
Using corresponding commit ids of hiprand for each releases

### PR DESCRIPTION
**Pull request by renjithravindrankannath**
Require corresponding commit ids of hiprand for each releases hiprand changes in master is conflicting older releases since the recent changes in master branch conflicts with older releases of rocrand.

### CODE REVIEW

**Comment by renjithravindrankannath** [comment]
@spackbot fix style

**Review by srekolam — DISMISSED** [review]
LGTM.  I have seen this failure, and this is due to hiprand code that is referred inside rocrand recipe is using develop branch from 5.1.0 : but newer changes for 5.4
.0 and later were merged recently to develop branch of hiprand and so it will fail for older releases.

File: var/spack/repos/builtin/packages/rocrand/package.py [hunk 1]
```diff
@@ -21,7 +21,7 @@ class Rocrand(CMakePackage):
 
     maintainers = ["cgmb", "srekolam", "renjithravindrankannath"]
     libraries = ["librocrand"]
-
+    version("master", branch="develop")
```
Comment (cgmb) [inline var/spack/repos/builtin/packages/rocrand/package.py]: I would suggest

    version("develop", branch="develop")
    version("master", branch="master")

File: var/spack/repos/builtin/packages/rocrand/package.py [hunk 2]
```diff
@@ -120,13 +120,30 @@ class Rocrand(CMakePackage):
...
+        )
     resource(
         name="hipRAND",
         git="https://github.com/ROCmSoftwarePlatform/hipRAND.git",
         branch="develop",
         destination="",
         placement="hiprand",
-        when="@5.1.0:",
+        when="@master",
     )
```
Comment (cgmb) [inline var/spack/repos/builtin/packages/rocrand/package.py]: For which you'd need a second one of these with `when="@develop"`.

**Review by cgmb — COMMENTED** [review]
It looks good, but I think we can do better with the added `@master` version.

File: var/spack/repos/builtin/packages/rocrand/package.py (hunk 1 above)
Comment (renjithravindrankannath) [reply var/spack/repos/builtin/packages/rocrand/package.py]: Thank you @cgmb.   Updated as per the suggestion.

**Review by cgmb — APPROVED** [review]

File: var/spack/repos/builtin/packages/rocrand/package.py (hunk 2 above)
Comment (renjithravindrankannath) [reply var/spack/repos/builtin/packages/rocrand/package.py]: Thank you..!

**Comment by renjithravindrankannath** [comment]
@tldahlgren can you please let us know if this can be merged ?

**Comment by cgmb** [comment]
Not sure who does the backports, but this should probably also be backported to v0.19.1.

File: var/spack/repos/builtin/packages/rocrand/package.py [hunk 3]
```diff
@@ -120,13 +121,38 @@ class Rocrand(CMakePackage):
...
     patch("hiprand_prefer_samedir_rocrand.patch", working_dir="hiprand", when="@5.2.0:")
 
+    # Add hiprand sources thru the below
+    for d_version, d_commit in [
+        ("5.3.0", "125d691d3bcc6de5f5d63cf5f5a993c636251208"),
+        ("5.2.3", "7190cc2199d8d97dbb6ce3fd179f1a62ad8fb13b"),
+        ("5.2.1", "7190cc2199d8d97dbb6ce3fd179f1a62ad8fb13b"),
+        ("5.2.0", "7190cc2199d8d97dbb6ce3fd179f1a62ad8fb13b"),
+        ("5.1.3", "7190cc2199d8d97dbb6ce3fd179f1a62ad8fb13b"),
+        ("5.1.0", "7190cc2199d8d97dbb6ce3fd179f1a62ad8fb13b"),
```
Comment (cgmb) [inline var/spack/repos/builtin/packages/rocrand/package.py]: How did you decide on these versions? I'd have expected them to match the commit id listed by the hipRAND submodule in rocRAND.

```
5.3.0: 12e2f070337945318295c330bf69c6c060928b9e
5.2.3: 12e2f070337945318295c330bf69c6c060928b9e
5.2.1: 12e2f070337945318295c330bf69c6c060928b9e
5.2.0: 12e2f070337945318295c330bf69c6c060928b9e
5.1.3: 20ac3db9d7462c15a3e96a6f0507cd5f2ee089c4
5.1.0: 20ac3db9d7462c15a3e96a6f0507cd5f2ee089c4
```

**Review by cgmb — CHANGES_REQUESTED** [review]
Sorry for changing this around, but I actually have a question about the commit hashes you chose.

File: var/spack/repos/builtin/packages/rocrand/package.py (hunk 3 above)
Comment (renjithravindrankannath) [reply var/spack/repos/builtin/packages/rocrand/package.py]: Hi @cgmb ,
These Commit IDs are from hiprand repo w.r.t the time stamps of 5.3.0 and 5.2.3 release tags of rocrand.
Commit IDs [125d691d3bcc6de5f5d63cf5f5a993c636251208 ](https://github.com/ROCmSoftwarePlatform/hipRAND/commit/125d691d3bcc6de5f5d63cf5f5a993c636251208) from the hipRAND repo (https://github.com/ROCmSoftwarePlatform/hipRAND) was taken based on the timestamp of release tag of 5.3.0.
Prior to the 5.3.0 related changes older release had not conflicts with the commit ID [7190cc2199d8d97dbb6ce3fd179f1a62ad8fb13b](https://github.com/ROCmSoftwarePlatform/hipRAND/commit/7190cc2199d8d97dbb6ce3fd179f1a62ad8fb13b).
Hence it is used for 5.1.0 through 5.2.3.

Comment (renjithravindrankannath) [reply var/spack/repos/builtin/packages/rocrand/package.py]: commit ids were picked based on the rocm releases(5.2.0, 5.2.3,5.3.0 ) tag's time stamp of the rocrand . I tried to match the respective commit id for the hiprand.

**Comment by renjithravindrankannath** [comment]
@cgmb,  Please let me know if its good to merge.

**Review by tldahlgren — DISMISSED** [review]
Confirmed the branches exist and appreciate that the version names match branches.

File: var/spack/repos/builtin/packages/rocrand/package.py (hunk 3 above)
Comment (cgmb) [reply var/spack/repos/builtin/packages/rocrand/package.py]: I don't think the timestamp is the correct way to choose the corresponding hipRAND version. Releases are made from a feature branch, so the development branch may have added commits intended for the next release at the time the previous release is tagged.

**Comment by cgmb** [comment]
I've asked @stanleytsang-amd to confirm the commit hashes. The specified versions do work together, but I don't think they are the intended correspondence. If this problem is disruptive, this PR could be merged as-is to get rocrand compiling again. However, I think it's likely we'd need to revisit the fix later if we did so.

**Comment by cgmb** [comment]
Please update to the commit ids listed in [my comment](https://github.com/spack/spack/pull/34545#discussion_r1050226138), as they have been confirmed with the upstream hipRAND team.

**Comment by renjithravindrankannath** [comment]
> Member
Updated the commit ids as commented. Thanks @cgmb

**Review by cgmb — APPROVED** [review]

**Comment by renjithravindrankannath** [comment]
@spackbot run pipeline

**Comment by renjithravindrankannath** [comment]
Hi @tldahlgren , Can you please let me know if we can merge this?

**Review by haampie — APPROVED** [review]

**Comment by renjithravindrankannath** [comment]
Thank you @haampie ..!

**Pull request merged**